From 1c90385d8012765d620c177c09d4155bfac397b8 Mon Sep 17 00:00:00 2001
From: Katarina Behrens <Katarina.Behrens@cib.de>
Date: Fri, 8 Jun 2018 15:04:13 +0200
Subject: [PATCH] Add overloaded read/send funcs for uint64_t

otherwise on 32bit platforms, no matching overloaded func is found
(I don't quite understand why, this template doom is beyond my
comprehension)

Change-Id: Ief4571ad735ad4efea9ddc70daea45885e5954c7
---
 vcl/unx/gtk3_kde5/filepicker_ipc_commands.hxx | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/vcl/unx/gtk3_kde5/filepicker_ipc_commands.hxx b/vcl/unx/gtk3_kde5/filepicker_ipc_commands.hxx
index 8ce8cfa9e78d..0d9e831dca46 100644
--- a/vcl/unx/gtk3_kde5/filepicker_ipc_commands.hxx
+++ b/vcl/unx/gtk3_kde5/filepicker_ipc_commands.hxx
@@ -100,6 +100,12 @@ inline void readIpcArg(std::istream& stream, sal_uIntPtr& value)
     stream.ignore(); // skip space
 }
 
+inline void readIpcArg(std::istream& stream, uint64_t value)
+{
+    stream >> value;
+    stream.ignore(); // skip space
+}
+
 inline void readIpcArgs(std::istream& /*stream*/)
 {
     // end of arguments, nothing to do
@@ -135,6 +141,8 @@ inline void sendIpcArg(std::ostream& stream, sal_Int16 value) { stream << value
 
 inline void sendIpcArg(std::ostream& stream, sal_uIntPtr value) { stream << value << ' '; }
 
+inline void sendIpcArg(std::ostream& stream, uint64_t value) { stream << value << ' '; }
+
 inline void sendIpcArgsImpl(std::ostream& stream)
 {
     // end of arguments, flush stream
-- 
2.17.1

